src/building/granary.c

```diff
@@ -141,6 +141,41 @@ int building_granary_add_resource(building *granary, int resource, int is_produc
     return 1;
 }
 
+int building_granaries_add_resource(int resource, int amount, int respect_settings)
+{
+    if (!resource_is_food(resource)) {
+        return amount;
+    }
+
+    int remaining_to_add = amount;
+    for (building *b = building_first_of_type(BUILDING_GRANARY); b; b = b->next_of_type) {
+        if (b->resources[RESOURCE_NONE] <= 0) {
```

Comment:
Don't forget to check for valid granaries:

```c
if (b->state != BUILDING_STATE_IN_USE) {
    continue;
}
```

Reply:
Done.